libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -776,8 +776,11 @@ func (n *bridgeNetwork) setPerPortIptables(b portBinding, enable bool) error {
 	if err := setPerPortNAT(b, v, proxyPath, bridgeName, enable); err != nil {
 		return err
 	}
-	if err := setPerPortForwarding(b, v, bridgeName, enable); err != nil {
-		return err
+
+	if !n.isGwModeUnprotected(v) {
```

Comment:
Really nit-picky, but I had to look 2 levels deep to see the exact details;  

- we need `isGwModeUnprotected` to select the gateway mode for either IPv4 or IPv6
- `n.config.GwModeIPv6` / `n.config.GwModeIPv4` are a `gwMode`, which is effectively a glorified string, but we started to dig in to adding `gwMode.something() bool` accessors, which just compare it

Wondering if it would make sense to have a `n.gatewayMode(v iptables.IPVersion) gwMode` method, which would allow comparing it where we need;


```go
if n.gateWayMode(v) == gwModeNATUnprot {
  // ...
}
```

or for cases where there's multiple code-paths;


```go
switch  n.gateWayMode(v) {
case gwModeNAT:
    // ....
case gwModeNATUnprot:
    // ...
}
```

Reply:
If we end up adding a `routed-unprotected` mode, this code will want to know whether the mode is `nat-unprotected` or `routed-unprotected` ... it could explicitly check for the mode being any one of the ones it's interested in - but I think collecting the logic into one place is better. Less danger of missing-a-bit when things change. But, can change it if you like?

Reply:
I've unwrapped it a bit ... now there's a `bridgeNetwork.gwMode()` that returns a `gwMode`, and the caller checks `unprotected()` themselves.

(Similarly, could replace `getNATDisabled()` with `gwMode()`, although it'd mean acquiring the mutex an extra time to get both modes.)